ReactiveXComponent/XComponentApi.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+
+namespace ReactiveXComponent
+{
+    public class XComponentApi : IXComponentApi
+    {
+        private bool _disposed;
+        private readonly Stream _file;
+        private readonly IXCSessionFactory _sessionFactory;
+
+        public XComponentApi(Stream file)
+        {
+            _file = file;
+            _sessionFactory = new XCSessionFactory();
+        }
+
+        public XCSession CreateSession()
+        {
+            return (_sessionFactory?.CreateSession(_file) as XCSession);
+        }
+
+        private void Close()
+        {
+            _file?.Close();
+            _sessionFactory?.Close();
+        }
+
+        private void Dispose(bool disposed)
```

Comment:
improve your IDisposable pattern
check this link https://msdn.microsoft.com/fr-fr/library/system.idisposable(v=vs.110).aspx